File: OOP/typescript/Inheritance/single.ts
// Parent Class
class Product {
  constructor(
    public skuID: string,
    public price: number,
    public manufacturer: string
  ) {}

  displayProduct() {
    console.log(`SKU: ${this.skuID}`);
    console.log(`Price: ${this.price}`);
    console.log(`Manufacturer: ${this.manufacturer}`);
  }
}

// Child Class
class PerishableProduct extends Product {
  constructor(
    skuID: string,
    price: number,
    manufacturer: string,
    public expirationDate: string,
    public storageTemperature: number
  ) {
    super(skuID, price, manufacturer);
  }

  displayPerishable() {
    this.displayProduct();
    console.log(`Expires On: ${this.expirationDate}`);
    console.log(`Storage Temp: ${this.storageTemperature}°C`);
  }
}

const milk = new PerishableProduct("SKU123", 50, "DairyCo", "2026-01-01", 4);
milk.displayPerishable();

File: OOP/java/Inheritance/Single.java
package Inheritance;
class Product {
    String ID;
    double price;
    String manufacturer;

    Product(String ID, double price, String manufacturer) {
        this.ID = ID;
        this.price = price;
        this.manufacturer = manufacturer;
    }

    void displayProduct() {
        System.out.println("SKU: " + ID);
        System.out.println("Price: " + price);
        System.out.println("Manufacturer: " + manufacturer);
    }
}

// Child Class
class PerishableProduct extends Product {
    String expirationDate;
    double storageTemperature;

    PerishableProduct(String ID, double price, String manufacturer,
                      String expirationDate, double storageTemperature) {
        super(ID, price, manufacturer);
        this.expirationDate = expirationDate;
        this.storageTemperature = storageTemperature;
    }

    void displayPerishable() {
        displayProduct();
        System.out.println("Expires On: " + expirationDate);
        System.out.println("Storage Temp: " + storageTemperature + "°C");
    }
}

public class Single {
    public static void main(String[] args) {
        PerishableProduct milk = new PerishableProduct("SKU123", 50.00, "DairyCo", "2026-01-01", 4.0);
        milk.displayPerishable();
    }
}
